### PR TITLE
doc: resturcture the doc folder

### PR DESCRIPTION
**Pull request by yati1998**
This commit is the restructure the doc folder to
include sub-folders like csi-addons, rbd, cephfs
to contain related docs and other general docs can be placed under doc folder.
This will enhance the doc structure will make it easier for the users to search the docs as it get more populated.

### CODE REVIEW

**Review by Madhu-1 — COMMENTED** [review]
we need to move the mount corruption doc to design and same for cephfs snapshot backed volumes as well and for rbd-nbd as well. 

we can have a deploy folder for deployments instead of having separate folder in the doc itself.

@Nikhil-Ladha @Rakshith-R any thoughts?

**Comment by nixpanic** [comment]
While moving the file into new directories per component, you can remove the redundant component name in the filename. For example:

docs/deploy-rbd.md → docs/rbd/deploy.md

**Comment by nixpanic** [comment]
> we can have a deploy folder for deployments instead of having separate folder in the doc itself.

I agree if you mean moving the files like this:

docs/deploy-rbd.md → deploy/rbd.md

Possibly add a symlink like docs/rbd/deploy.md pointing to the deploy/rbd.md file.

**Comment by Nikhil-Ladha** [comment]
Moving to deploy folder and having symlink sounds good!
This way we will have better accessibility.

**Comment by Rakshith-R** [comment]
> > we can have a deploy folder for deployments instead of having separate folder in the doc itself.
> 
> I agree if you mean moving the files like this:
> 
> docs/deploy-rbd.md → deploy/rbd.md
> 
> Possibly add a symlink like docs/rbd/deploy.md pointing to the deploy/rbd.md file.


sounds good to me

File: deploy/rbd/deploy.md [hunk 1]
```diff
@@ -0,0 +1 @@
+docs/rbd/deploy.md
```
Comment (Madhu-1) [inline deploy/rbd/deploy.md]: accessing symlink failed for me, can you please cross check this and see if you are able to access the file if you open it from deploy folder?

Comment (yati1998) [reply deploy/rbd/deploy.md]: yes, shows error while opening, will check on that

Comment (yati1998) [reply deploy/rbd/deploy.md]: it works now, please do review it again

**Comment by yati1998** [comment]
\retest

**Review by Madhu-1 — APPROVED** [review]

**Comment by Madhu-1** [comment]
@mergifyio rebase

**Review by iPraveenParihar — APPROVED** [review]

**Comment by nixpanic** [comment]
@mergifyio queue

**Pull request merged**